app/main.py

```diff
@@ -1,13 +1,45 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            }
+            for error in errors
+        ],
+
+        "path": file_path,
+        "status": "passed"
+        if not any(err["filename"] == file_path for err in errors) else
+        "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [{
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            }
+            for error in linter_report[file_path]
+        ],
+        "path": file_path,
+        "status": "passed" if not linter_report[file_path] else "failed"
+    } for file_path in linter_report]
```

Comment:
You should follow the `DRY` principle as mentioned [in the checklist](https://github.com/mate-academy/py-linter-formatter/blob/master/checklist.md), there is code duplication with the `format_single_linter_file` function here. Use `format_single_linter_file` to avoid it.